### PR TITLE
[REG-2008] Overhaul recording, replay, and gameplay session docs

### PR DESCRIPTION
**Pull request by abeizer**
* Created new page specifically for recording + replaying recordings
* Gameplay Session is no longer synonymous with local recording
  * Gameplay Session has one page in the sidebar now
* Updated screenshots to show RG Toolbar icons from new `RGOverlayCanvas`
* Created an accordion component for hiding content until the user chooses to expand it
  * Content can be rendered as markdown. someone explain to me why docusaurus doesn't support rendering markdown from components out of the box :/
* Added feedback link to the site footer

https://www.loom.com/share/2f4d425f1d6a49ef868f0bc05efec08d?sid=a8027e98-5ac1-4d19-ae92-eb88efc4d92d

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: docs/core-concepts/recording-gameplay.mdx [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+#### Manual Replay
+
+Press the **Menu** icon in the RG Toolbar while in "Play" mode to select a recording to replay.
+This will launch a file explorer at the location of the [recordings directory](#recorded-data-formats) on your device.
+Navigate into the folder of the recording you'd like to replay, select its `bot_segments.zip` file, and then click **Load Replay**.
+
+![Select a Recording](img/recording-gameplay/select-recording.png)
+
+<div className="container row text--center">
```
Comment (vontell) [inline docs/core-concepts/recording-gameplay.mdx:139]: I like the styling here!

File: docs/core-concepts/recording-gameplay.mdx [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+references in your test assembly to use the `RGTestUtils` class.
+
+<img src={TestAssemblyRefs} alt="Test assembly references for Regression Games SDK" style={{display: "block", margin: "auto"}}/>
+:::
+
+:::tip
+By default, [recordings are saved outside your project](#recorded-data-formats).
+Make sure you copy the recordings you'd like to replay into your project
+to ensure they'll be available on other machines that run these automated tests.
+:::
```
Comment (vontell) [inline docs/core-concepts/recording-gameplay.mdx:200]: Hopefully this will be gone soon!

File: docs/core-concepts/recording-gameplay.mdx [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+        yield return RGTestUtils.WaitForScene("MainMenu");
+
+        // Start playback
+        // This will automatically create a new recording of the replay
+        PlaybackResult playbackResult = null;
+        yield return RGTestUtils.StartPlaybackFromZipFile(recordingPath, result => playbackResult = result);
+
+        // Print out the recording path for viewing later
+        RGDebug.LogInfo("Played back and recorded the test run - saved to " + playbackResult.saveLocation);
+        Assert.IsNotNull(playbackResult.saveLocation);
```
Comment (vontell) [inline docs/core-concepts/recording-gameplay.mdx:243]: This was a bug in my code - these should be swapped, and we need to add another assert. There are situations where playbackResult is null.

```suggestion
        Assert.IsNotNull(playbackResult);
        Assert.IsNotNull(playbackResult.saveLocation);
        RGDebug.LogInfo("Played back and recorded the test run - saved to " + playbackResult.saveLocation);

```

**Review by vontell — CHANGES_REQUESTED** [review]
Just one small change, this looks awesome!

**Review by batu — APPROVED** [review]
This reads really well. Appreciate the accordions and table styling

File: docs/core-concepts/recording-gameplay.mdx (hunk 3 above)
Comment (abeizer) [reply docs/core-concepts/recording-gameplay.mdx]: Accepted your suggestion, but can you elaborate on what situations cause the playbackResult to be null? Is this when playback encounters some error?

**Review by addisonbgross — APPROVED** [review]

File: docs/core-concepts/recording-gameplay.mdx (hunk 3 above)
Comment (vontell) [reply docs/core-concepts/recording-gameplay.mdx]: Exactly, this happens when the playback fails... so this is kind of actually a bug in our own system, but I haven't had time to dig deeper into this yet.

**Review by vontell — APPROVED** [review]
